### PR TITLE
Add docker's add/drop capabilites

### PR DESCRIPTION
**Pull request by hustcat**
Add docker's add/drop capabilites

### CODE REVIEW

**Comment by dchen1107** [comment]
Thanks you very much for your contribution. Are you still actively working on it? or plan to do so? If yes, could you please ping me back? We want to maintain a list of active PRs; if I didn't hear from you by next Monday (2/2/2015), I plan to close it. I think this PR is LGTM in general with a rebase though. Thanks!

**Comment by bgrant0607** [comment]
I would like to get this merged. Could you please rebase?

I'm willing to forego the documentation for now. I can't even find comparable documentation on Docker's site.

**Comment by hustcat** [comment]
@dchen1107 @bgrant0607 Sorry about the delay, I have rebased it.

**Comment by hustcat** [comment]
I don't why failed on Shippable, and I can't even open the details page.

@bgrant0607 I added some descriptive information here, but I do not know where to put better.

Capabilities
------
By default, Docker containers are "unprivileged" and cannot, for example, run a Docker daemon inside a Docker container. We can have fine grain control over the capabilities using cap-add and cap-drop.More details [here](https://docs.docker.com/reference/run/#runtime-privilege-linux-capabilities-and-lxc-configuration).

The relationship between Docker's capabilities and [Linux capabilities](http://man7.org/linux/man-pages/man7/capabilities.7.html)

| Docker's capabilities | Linux capabilities |
| ---- | ---- |
| SETPCAP |  CAP_SETPCAP |
| SYS_MODULE |  CAP_SYS_MODULE |
| SYS_RAWIO |  CAP_SYS_RAWIO |
| SYS_PACCT |  CAP_SYS_PACCT |
| SYS_ADMIN |  CAP_SYS_ADMIN |
| SYS_NICE |  CAP_SYS_NICE |
| SYS_RESOURCE |  CAP_SYS_RESOURCE |
| SYS_TIME |  CAP_SYS_TIME |
| SYS_TTY_CONFIG |  CAP_SYS_TTY_CONFIG |
| MKNOD |  CAP_MKNOD |
| AUDIT_WRITE |  CAP_AUDIT_WRITE |
| AUDIT_CONTROL |  CAP_AUDIT_CONTROL |
| MAC_OVERRIDE |  CAP_MAC_OVERRIDE |
| MAC_ADMIN |  CAP_MAC_ADMIN |
| NET_ADMIN |  CAP_NET_ADMIN |
| SYSLOG |  CAP_SYSLOG |
| CHOWN |  CAP_CHOWN |
| NET_RAW |  CAP_NET_RAW |
| DAC_OVERRIDE |  CAP_DAC_OVERRIDE |
| FOWNER |  CAP_FOWNER |
| DAC_READ_SEARCH |  CAP_DAC_READ_SEARCH |
| FSETID |  CAP_FSETID |
| KILL |  CAP_KILL |
| SETGID |  CAP_SETGID |
| SETUID |  CAP_SETUID |
| LINUX_IMMUTABLE |  CAP_LINUX_IMMUTABLE |
| NET_BIND_SERVICE |  CAP_NET_BIND_SERVICE |
| NET_BROADCAST |  CAP_NET_BROADCAST |
| IPC_LOCK |  CAP_IPC_LOCK |
| IPC_OWNER |  CAP_IPC_OWNER |
| SYS_CHROOT |  CAP_SYS_CHROOT |
| SYS_PTRACE |  CAP_SYS_PTRACE |
| SYS_BOOT |  CAP_SYS_BOOT |
| LEASE |  CAP_LEASE |
| SETFCAP |  CAP_SETFCAP |
| WAKE_ALARM |  CAP_WAKE_ALARM |
| BLOCK_SUSPEND |  CAP_BLOCK_SUSPEND |

File: pkg/api/v1beta1/types.go [hunk 1]
```diff
@@ -241,6 +241,16 @@ const (
 	PullIfNotPresent PullPolicy = "PullIfNotPresent"
 )
 
+type CapabilityType string
+
+// Capabilities represent POSIX capabilities that can be added or removed to a running container.
+type Capabilities struct {
+	// Added capabilities
+	Add []CapabilityType `json:"add,omitempty"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Please add descriptions to the fields. All fields must have descriptions in v1beta1/types.go and v1beta2/types.go

File: pkg/api/v1beta1/types.go [hunk 2]
```diff
@@ -267,6 +277,8 @@ type Container struct {
 	Privileged bool `json:"privileged,omitempty" description:"whether or not the container is granted privileged status; defaults to false"`
 	// Optional: Policy for pulling images for this container
 	ImagePullPolicy PullPolicy `json:"imagePullPolicy" description:"image pull policy; one of PullAlways, PullNever, PullIfNotPresent; defaults to PullAlways if :latest tag is specified, or PullIfNotPresent otherwise"`
+	// Optional: Capabilities for container.
+	Capabilities Capabilities `json:"capabilities,omitempty" description:" capabilities for container"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Please remove the leading space from the description, here and elsewhere.

**Comment by bgrant0607** [comment]
@hustcat Thanks for writing that up. Could you please create docs/containers.md, and put the documentation there? We can add more general container documentation there in the future.

I can't see the shippable results, either. :-(

**Comment by hustcat** [comment]
@bgrant0607 Thank you for the advise, please re-review it.

File: pkg/api/v1beta1/types.go (hunk 1 above)
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: We have a presubmit check that verifies that API fields have descriptions. I can't merge this until they are added.

Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Specifically, Add and Drop still need descriptions, in v1beta1/types.go and in v1beta2/types.go.

**Comment by bgrant0607** [comment]
Thanks @hustcat. Please add the Add and Drop field descriptions, rebase, and squash the commits, and then I'll merge this.

**Comment by hustcat** [comment]
@bgrant0607 Sorry about my negligence. I have squashed my commits.Thanks.

**Comment by bgrant0607** [comment]
LGTM. Thanks!

**Pull request merged**